### DISCUSSION TITLE
Signal problems on the green line

### DISCUSSION THREAD

**Discussion by czsmith** [Dilbert66/esphome-vistaECP#124] [Q&A]
I built the ESP8266 version using the single opto exactly as shown in the first diagram. (Well, maybe I used a 34K resistor instead of a 33K, but that's about it.)

I am have been getting "Check 107" errors and errors about zones on my expansion card with the device.

What is particularly interesting is that if I do not connect U2/pin 5 (collector) to the red signal, I have absolutely no problem and it runs for days.  If I do connect it, I  get the errors within minutes.

So I put a scope on the U2 input (pin 1) with the yellow probe and the output (pin 5) with the blue.  Something is amiss about the signal... the output rises before the driving input. And the output can remain high while the driving signal is low.

<img width="1058" alt="Screen Shot 2024-02-21 at 11 35 38 AM" src="https://github.com/Dilbert66/esphome-vistaECP/assets/664334/6b172fbb-4e6e-49e5-8ff5-d1e381118a49">

This looks very much like there is a collision. I suspect:
- An outside source is driving the green line (blue on scope) line high while the U2 input is low.  
- Output is high when both green and input signals are high
- Output is low if the green signal is low, regardless of the driving signal.

I am not doing anything on the HA alarm panel to trigger a transmission (e.g. no keypresses) I suspect that there is polling or checking going to that causes the 8266 to start transmitting almost immediately after there's activity on the green line.  Or just very frequent coincidences.

To check this out a bit more, I added a 4.7K resister between the red input that's the pull-up on the opto (U2/pin 5).  The pattern looks much clearer then.

![overwriteing](https://github.com/Dilbert66/esphome-vistaECP/assets/664334/485aceed-ec14-4d88-a212-ed0217da2527)

The intermediate voltage shows the 8266 is attempting to pull the green signal high but someone else is driving it low.

I have set my device address to 17 and the only devices on the bus are:
- Main Vista 20p board
- Single 4219 expansion board set to address 07 for zones 9-16
- Honeywell 6160RF alpha keypad at address 16
- ESPHome-Vista board set with an keypad address of 17

Is this board communicating even when not pressing buttons or activating the alarm?

Any idea what's going on or areas that I should be checking next?

I tried to add the VistaAlarm yaml file, but it cannot be attached and if quoted is a display mess.  You can find it here on Google Drive: https://drive.google.com/file/d/10dy3p6kAemYUvefYIl2dgYlmd-v3dxOi/view?usp=sharing

**Comment by Dilbert66** [comment]
It sounds  to me like you have a wiring issue.  The circuit design and firmware are fine.  Also make sure that the panel ground is connected to the ESP ground as well as that could cause issues if missing.  Otherwise recheck your wiring. Make sure you are connected to the right pins, etc. Not much i can offer.

As to your question reqarding sending when no keys are sent, yes it does send an "*" once when it sees the "press * " message so that it can get zone data. That would not cause the issues you have. It does not send continuously.

**Reply by czsmith** [reply]
Wiring is all good.

What I did notice is that I did not disable the expanderaddr1.  It was still "07" from the base yaml file I got from github.

If that is the case, and I have a real expansion module at the same address, would it make sense that both my board and the real expansion board are being polled and the replies are stepping on each other?

(I broke something in my yaml file so I can't try that right now.)

I really appreciate the quick response and am finding that this project is a lot of fun.  As a retired EE, I haven't had cause to prototype anything in quite a while.

**Reply by czsmith** [reply]
Well, I got my yaml syntax errors fixed.

WIth expanderaddr1 set to "0", it works perfectly. There were definitely messages from the real expander board and the ecphome-vistaalarm board stomping on each other.  Guess I didn't pay enough attention to the difference between "emulate" and "monitor" in your sample yaml comments.

Thanks for this great project.

**Comment by Dilbert66** [comment]
Yes they would conflict. Set the expanders to 0. I didnt notice the expander line in your yaml otherwise I might have mentioned it. Glad you got it sorted!
